cip/CIP2016-01-26-mandatory-match.adoc

```diff
@@ -0,0 +1,74 @@
+= CIP2016-01-26 - Mandatory Match
+:numbered:
+:toc:
+:toc-placement: macro
+:source-highlighter: codemirror
+
+*Author:* Stefan Plantikow <stefan.plantikow@neotechnology.com>
+
+[abstract]
+.Abstract
+--
+This CIP proposes adding a new variant of the `MATCH` clause called `MANDATORY MATCH` that fails when it does not match at least a single row in the underlying graph.
+--
+
+toc::[]
+
+== Motivation
+
+When writing applications using Cypher, many queries will start off by looking up multiple individual nodes based on an unique id with the implied expectation of matching exactly one node. When that expectation is not fulfilled (either due to sending a wrong id or the node having been deleted) it becomes important to identify which lookup failed. Achieving  this currently is not well supported by Cypher.
+
+== Background
+
+The idea for this CIP grew out of a (now abandoned) dog fooding project for storing benchmarking information in Neo4j.
```

Comment:
I actually feel that these details are unnecessary. What would be relevant is describing what problem this proposal solves (the previous section already does that) and when that problem arises.